e2e-tests/src/test/treasury.rs

```diff
@@ -38,11 +58,16 @@ pub fn channeling_fee(config: Config) -> anyhow::Result<()> {
         treasury_balance_after, issuance_after
     );
 
-    assert!(
-        issuance_after <= issuance_before,
-        "Unexpectedly {} was minted",
-        issuance_after - issuance_before
-    );
+    if issuance_after > issuance_before {
+        let difference = issuance_after - issuance_before;
+        assert_eq!(
+            difference % possibly_treasury_gain_from_staking,
+            0,
+            "Unexpectedly {} was minted, and it's not related to staking treasury reward which is {}",
+            difference, possibly_treasury_gain_from_staking
+        );
```

Comment:
Shouldn't this also be checked using `check_treasury_balance`? Or a similar function if this one is not appropriate?

Reply:
Let me extract method for this one too.